In [15]:
from scipy import stats
import pandas as pd
import numpy as np

In [16]:
#Data
data = {'Print-heads':[1.13,1.55,1.43,0.92,1.25,1.36,1.32,0.85,1.07,1.48,1.20,1.33,1.18,1.22,1.29]}

In [17]:
sampled_data = pd.DataFrame(data)

In [18]:
sampled_data

,Print-heads
0,1.13
1,1.55
2,1.43
3,0.92
4,1.25
5,1.36
6,1.32
7,0.85
8,1.07
9,1.48


In [19]:
sampled_data.mean()

Print-heads    1.238667
dtype: float64

In [20]:
sampled_data.std()

Print-heads    0.193164
dtype: float64

In [21]:
#a.Building 99% confidence interval using sample standard deviation:
sampled_data_ci=stats.t.interval(0.99,15,
                                  loc = sampled_data.mean(),
                                  scale = sampled_data.std())
print(sampled_data_ci)
print( 'Mean at 99% confidence interval is:', np.round(sampled_data_ci, 4))

(array([0.66946744]), array([1.8078659]))
Mean at 99% confidence interval is: [[0.6695]
 [1.8079]]


The main reason behind using the t_distribution was the sigma /standard deviation is unknown to this.using t_distribution we can also build the 99% confidence interval using the sample standars deviation.to build this confidence interval,we use sample data mean,standard deviation and thet_distribution.

In [22]:
#b.Building 99% confidence interval using known population standard deviation:
population_std = 0.2
sampled_data_ci=stats.norm.interval(0.99,loc =sampled_data.mean(),
                                  scale =sampled_data.std())
print(sampled_data_ci)
print( 'Mean at 99% confidence interval is:', np.round(sampled_data_ci, 4))

(array([0.74110884]), array([1.73622449]))
Mean at 99% confidence interval is: [[0.7411]
 [1.7362]]


CONCLUSION :-
(1) If a manufacturer of print_heads for personal computers is interesting in estimating the mean durability of their print_heads in terms of the numbers of the characters printed before failure,SO the 99% confidence interval using sample standard deviation for the mean number of characters printed before failure will be in the range of [[0.6695] to [1.8079]](million).

(2) 99% confidence interval using known population standard deviation the range is [[0.7411] to [1.7362]](million),
                       AND WE ARE 99% CONFIDEDT ABOUT THIS ANSWER!